receiver/postgresqlreceiver/scraper.go

```diff
@@ -159,18 +157,18 @@ func (p *postgreSQLScraper) scrape(ctx context.Context) (pmetric.Metrics, error)
 		}
 		defer dbClient.Close()
 		numTables := p.collectTables(ctx, now, dbClient, database, &errs)
-
+		p.retrieveDBMetrics(ctx, dbClient, databases, r, &errs)
 		p.recordDatabase(now, database, r, numTables)
 		p.collectIndexes(ctx, now, dbClient, database, &errs)
 		p.collectFunctions(ctx, now, dbClient, database, &errs)
+		p.collectBGWriterStats(ctx, now, dbClient, &errs)
```

Comment:
Most of these metrics are cluster level and not DB level. Do you think it makes sense to move them in the database loop ?

Reply:
I can add an `if` to only take the first database from `databases` to avoid the loop.

I need to look each metrics produce to be sure all are cluster level metrics.

Reply:
@ishleenk17 
Except `collectDatabaseLocks` the other one are indeed cluster level metrics.
I can extract it from the database loop, and execute it on the first database.

When looking to see the actual code of `collectDatabaseLocks`, I think a patch it's possible to use the `database OID` column as filter.

The query is :
```sql
SELECT relname AS relation, mode, locktype,COUNT(pid)
	AS locks FROM pg_locks
	JOIN pg_class ON pg_locks.relation = pg_class.oid
	GROUP BY relname, mode, locktype;
```

I have not tested yet, but it seems `collectDatabaseLocks` contains the locks of every database.
https://www.postgresql.org/docs/17/view-pg-locks.html